Make fails to compile zmq code

On Ubuntu 14.04.4 LTS trusty.

I followed the instruction on .travis.yml

`sudo apt-get install libzmq3-dev`
`wget https://raw.githubusercontent.com/zeromq/cppzmq/master/zmq.hpp`

`lorenzod@lorenzod-hydra-node-01:~/src/hydra/src/main/c/zmq⟫ dpkg --get-selections | grep -v deinstall | grep zmq`
libzmq3:amd64                                   install
libzmq3-dev:amd64                               install

With latest zmq.hpp 
`lorenzod@lorenzod-hydra-node-01:~/src/hydra/src/main/c/zmq⟫ md5sum zmq.hpp` 
27b2b17a2800b80bef2b0248124ff979  zmq.hpp

Make fails:
`lorenzod@lorenzod-hydra-node-01:~/src/hydra/src/main/c/zmq⟫ make`
g++ -std=c++0x -g -I. hdaemon.pb.cc zmq_pub.cc -lzmq -lprotobuf -o ../../scripts/zmq_pub
zmq_pub.cc: In function 'int main()':
zmq_pub.cc:201:5: error: cannot convert 'zmq::socket_t' to 'void*' in initialization
     };
     ^
make: *** [../../scripts/zmq_pub] Error 1

I tried to force the cast to void * (i.e. static_cast<void *>()), but both executables fail.
`lorenzod@lorenzod-hydra-node-01:~/src/hydra/src/main/scripts⟫ ./zmq_sub`
terminate called after throwing an instance of 'std::logic_error'
  what():  basic_string::_S_construct null not valid
Aborted (core dumped)

--- a/src/main/c/zmq/zmq_pub.cc
+++ b/src/main/c/zmq/zmq_pub.cc
@@ -197,7 +197,7 @@ int main (void) {
   char message_ch[120];
   while(true) {
     zmq_pollitem_t items[] = {
-        {socket_rep, 0, ZMQ_POLLIN, 0}
+      {static_cast<void *>(socket_rep), 0, ZMQ_POLLIN, 0}
     };
     int rc = zmq_poll(items, 1, 0);
     if (rc == -1) {

--- a/src/main/c/zmq/zmq_sub.cc
+++ b/src/main/c/zmq/zmq_sub.cc
@@ -137,8 +137,8 @@ int main (int argc, char** argv) {
   char message_ch[120];
   while(true) {
     zmq_pollitem_t items[] = {
-      {socket_rep, 0, ZMQ_POLLIN, 0},
-      {socket_sub, 0, ZMQ_POLLIN, 0}
+      {static_cast<void *>(socket_rep), 0, ZMQ_POLLIN, 0},
+      {static_cast<void *>(socket_sub), 0, ZMQ_POLLIN, 0}
     };
     int rc = zmq_poll(items, 2, -1);
     if (rc == -1) {

